crates/sui-adapter/src/execution_engine.rs

```diff
@@ -358,12 +359,29 @@ fn advance_epoch<S: BackingPackageStore + ParentSync + ChildObjectResolver>(
     }
 
     for (version, modules) in change_epoch.system_packages.into_iter() {
-        let modules = modules
+        let modules: Vec<_> = modules
             .into_iter()
             .map(|m| CompiledModule::deserialize(&m).unwrap())
             .collect();
 
-        let new_package = Object::new_system_package(modules, version, tx_ctx.digest())?;
+        assert_invariant!(
+            !modules.is_empty(),
+            "System package must have at least one module"
+        );
+
+        let pkg_id = ObjectID::from(*modules[0].address());
+        let mut dependencies = vec![];
+        // Sui framework package has one dependency while Move stdlib package (the other one) has
+        // none.
+        // TODO: Should we assert that there could only be two system package here to avoid
+        // potential surprises with passing the right type of dependencies?
```

Comment:
Should `change_epoch` carry this information?

Reply:
Yeah, we don't want to assert here, because it will halt the network (very bad), and the binary running this transaction may not know about a hypothetical new framework package and its dependencies -- framework upgrade handling is a bit funny.  What we have here is fine, and I'll add the full support on top (it is along the lines of what Todd mentioned).